lib/schema.js

```diff
@@ -659,9 +692,50 @@ Schema.prototype.add = function add(obj, prefix) {
     }
   }
 
-  const addedKeys = Object.keys(obj).
-    map(key => prefix ? prefix + key : key);
-  aliasFields(this, addedKeys);
+  const aliasObj = Object.fromEntries(
+    Object.entries(obj).map(([key]) => ([prefix + key, null]))
+  );
+  aliasFields(this, aliasObj);
+  return this;
+};
+
+/**
+ * Add an alias for `path`. This means getting or setting the `alias`
+ * is equivalent to getting or setting the `path`.
+ *
+ * #### Example:
+ *
+ *     const toySchema = new Schema({ n: String });
+ *
+ *     // Make 'name' an alias for 'n'
+ *     toySchema.alias('n', 'name');
+ *
+ *     const Toy = mongoose.model('Toy', toySchema);
+ *     const turboMan = new Toy({ n: 'Turbo Man' });
+ *
+ *     turboMan.name; // 'Turbo Man'
+ *     turboMan.n; // 'Turbo Man'
+ *
+ *     turboMan.name = 'Turbo Man Action Figure';
+ *     turboMan.n; // 'Turbo Man Action Figure'
+ *
+ *     await turboMan.save(); // Saves { _id: ..., n: 'Turbo Man Action Figure' }
+ *
+ *
+ * @param {String} path real path to alias
+ * @param {String|String[]} alias the path(s) to use as an alias for `path`
+ * @return {Schema} the Schema instance
+ * @api public
+ */
+
+Schema.prototype.alias = function alias(path, alias) {
+  if (Array.isArray(alias)) {
+    for (const a of alias) {
+      this.alias(path, a);
+    }
```

Comment:
with the updates, couldnt this be changed to just `aliasFields(this, { [path]: alias })`? (or combine array and non-array paths)?